docs/reference/fields.txt

```diff
@@ -1079,3 +1081,70 @@ you can include the short versions of the modules.
      include Mongoid::Document
      include Mongoid::Timestamps::Updated::Short # For u_at only.
    end
+
+
+.. _field-names-with-periods-and-dollar-signs:
+
+Field Names with Periods (``.``) and Dollar Signs (``$``)
+=========================================================
+
+Mongo 5.0 `adds specialized support
+<https://docs.mongodb.com/manual/core/dot-dollar-considerations/#std-label-crud-concepts-dot-dollar-considerations>`_
+for field names that are prefixed with a dollar sign (``$``) or
+that contain periods (``.``).
+
+Fields prefixed with ``$`` or containing ``.`` are not directly accessible in standard queries.
+They may only be accessed with special database helper methods (new in MongoDB 5.0)
+including ``$getField``, ``$setField``, and ``$literal``.
+In normal usage scenarios, such fields behave as "hidden" fields on the document.
+In addition, they come with severe limitations, such as the inability
+to be indexed.
+
+Support in Mongoid
+------------------
+
+This section outlines the behavior of Mongoid and the Mongo Ruby Driver
+when working with fields whose names are prefixed with ``$`` or contain ``.``
+("specially-named fields"):
+
+* Mongoid's :ref:`field <field-types>` macro does not support defining specially-named fields.
+* Mongoid does not support getting or setting the values of specially-named fields.
+* Mongoid does not support queries or operations which reference specially-named fields.
```

Comment:
I don't see how this is true. If you are using 5.0 server you should be able to query these fields using the respective server operators.

Reply:
Does Mongoid support this today?

Reply:
What I meant was that you should be able to use the 5.0 server operators via the MQL syntax. Mongoid supports this (I would say) because it passes through MQL syntax to the driver and ultimately the server.

Mongoid doesn't presently provide *helpers* for the new 5.0 query operators, but a user utilizing Mongoid should be able to perform these queries hence I'd rather not state that something is not possible when there is a way to do it.

Reply:
I've clarified the language "Mongoid does not provide helper functions or idiomatic support..."